### PR TITLE
Fix exit crash in the plugin layer registry deconstructor (fixes #30681)

### LINKED ISSUE #30681: QGIS 3.80 Crashed
## User Feedback

just open and  exit QGIS. Nothting done with in tis period

## Report Details

*Crash ID*: b11db3d431a36dd8fdf085fe4349919d50f378d8

*Stack Trace*
<pre>
QMapNodeBase::nextNode :
QgsPluginLayerRegistry::~QgsPluginLayerRegistry :
QgsValidityCheckResult::QgsValidityCheckResult :
QgsApplication::~QgsApplication :
main :
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>

*QGIS Info*
QGIS Version: 3.8.0-Zanzibar
QGIS code revision: 11aff65f10
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 2.4.1
Running against GDAL: 2.4.1

*System Info*
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 6.1.7601

### PR DESCRIPTION
**Pull request by nirvn**
Deleting map items while tterating through that map == crasher.

### CODE REVIEW

File: src/core/qgspluginlayerregistry.cpp [hunk 1]
```diff
@@ -57,10 +57,10 @@ QgsPluginLayerRegistry::~QgsPluginLayerRegistry()
   if ( !mPluginLayerTypes.isEmpty() )
```
Comment (m-kuhn) [inline src/core/qgspluginlayerregistry.cpp]: Isn't this check redundant?

**Review by m-kuhn — APPROVED** [review]

File: src/core/qgspluginlayerregistry.cpp (hunk 1 above)
Comment (nirvn) [reply src/core/qgspluginlayerregistry.cpp]: Indeed.

Comment (m-kuhn) [reply src/core/qgspluginlayerregistry.cpp]: I've got my finger ready on the merge button, better remove this statement before the button's clicked ;)

Comment (nirvn) [reply src/core/qgspluginlayerregistry.cpp]: Heh. I'm not anywhere close to my workstation until tomorrow morning. Feel free to add a code suggestion I can accept, or lift your finger for 14 hours ;)

Comment (m-kuhn) [reply src/core/qgspluginlayerregistry.cpp]: 3 suggestions for a single change seems a lot. I've convinced my finger to snooze for now

**Comment by m-kuhn** [comment]
I just realized another blocker here, i.e. [deconstructors block entrances based on their description.](https://wowwiki.fandom.com/wiki/XT-002_Deconstructor), whereas destructors are some geeky constructs that get exectued in OOP when an object is being deleted.

**Comment by nirvn** [comment]
@m-kuhn , I humbly disagree, I believe the deconstructor should stay and be featured as a key highlight of upcoming 3.10 :wink:

**Comment by m-kuhn** [comment]
*QGIS 3.10 finally supporting 2-factor authentication: Enter your password **and** beat the deconstructor.*

**Comment by nirvn** [comment]
@m-kuhn , actually, the check isn't redundant, it's still required to conditionally output the debug msg.

I've updated the commit message to get rid of the fantasy character reference, will merge once travis is green :)

**Comment by m-kuhn** [comment]
>  it's still required to conditionally output the debug msg.

yes, I'd just have removed that along ;)

**Comment by nirvn** [comment]
I'm not sure but I think this debug msg is useful when debugging plugins.

**Comment by m-kuhn** [comment]
Let's leave it in there, it doesn't hurt :-)

**Pull request merged**